### PR TITLE
FIX: automatically convert fwd if necessary

### LINKED ISSUE #4545: API/BUG make_inverse_operator with depth=None
The logic of `make_inverse_operator` is a bit broken, we can't automatically use it with `loose > 0` and `depth=None` if the forward is not `surf_orig=True`

e.g.

```python
inv = make_inverse_operator(epochs.info, fwd, cov, loose=0.2, depth=.8)  # works
inv = make_inverse_operator(epochs.info, fwd, cov, loose=0.2, depth=None)  # does not work
```

The error mentions the issue with `loose` although the only change is in `depth`.

Instead, we need to do this first (although with different options, this is done automatically by make_inverse_operator in other cases)
```python
from mne.forward import convert_forward_solution
fwd = convert_forward_solution(fwd, surf_ori=True)
```

Sorry for the message brievity, I m not on my computer

JR

### PR DESCRIPTION
**Pull request by kingjr**
The forward model was automatically converted in other scenari, I guess it'd make sense to do it for all.

Closes #4545.

@agramfort Can you give it a look?

### CODE REVIEW

File: mne/minimum_norm/inverse.py [hunk 1]
```diff
@@ -1327,17 +1327,14 @@ def make_inverse_operator(info, forward, noise_cov, loose=0.2, depth=0.8,
                              'forward solution to do depth weighting even '
                              'when calculating a fixed-orientation inverse.')
         if not forward['surf_ori']:
-            forward = convert_forward_solution(forward, surf_ori=True)
+            forward = convert_forward_solution(forward.copy(), surf_ori=True)
```
Comment (larsoner) [inline mne/minimum_norm/inverse.py]: convert_forward_solution already has `copy=True` default

File: mne/minimum_norm/inverse.py [hunk 2]
```diff
@@ -1327,17 +1327,16 @@ def make_inverse_operator(info, forward, noise_cov, loose=0.2, depth=0.8,
...
         if not (0 <= loose <= 1):
             raise ValueError('loose value should be smaller than 1 and bigger '
                              'than 0, or None for not loose orientations.')
         if loose < 1 and not forward['surf_ori']:
-            raise ValueError('Forward operator is not oriented in surface '
-                             'coordinates. A loose inverse operator requires '
-                             'a surface-based, free orientation forward '
-                             'operator.')
+            forward = convert_forward_solution(forward, surf_ori=True,
+                                               copy=True)
```
Comment (agramfort) [inline mne/minimum_norm/inverse.py]: can you avoid the duplication of code to do the conversation. I am sure you can do it higher in the function.

File: mne/minimum_norm/inverse.py [hunk 3]
```diff
@@ -1327,17 +1327,16 @@ def make_inverse_operator(info, forward, noise_cov, loose=0.2, depth=0.8,
...
         assert forward['surf_ori']
     if loose is not None:
         if not (0 <= loose <= 1):
             raise ValueError('loose value should be smaller than 1 and bigger '
                              'than 0, or None for not loose orientations.')
         if loose < 1 and not forward['surf_ori']:
-            raise ValueError('Forward operator is not oriented in surface '
-                             'coordinates. A loose inverse operator requires '
-                             'a surface-based, free orientation forward '
-                             'operator.')
```
Comment (agramfort) [inline mne/minimum_norm/inverse.py]: can you add a logging message?

File: mne/minimum_norm/tests/test_inverse.py [hunk 4]
```diff
@@ -389,9 +389,11 @@ def test_make_inverse_operator_vector():
     inv_2 = make_inverse_operator(evoked.info, fwd_surf, noise_cov, depth=None)
     inv_3 = make_inverse_operator(evoked.info, fwd_surf, noise_cov, fixed=True,
                                   loose=None)
+    inv_4 = make_inverse_operator(evoked.info, fwd_fixed, noise_cov,
+                                  loose=.2, depth=None)
 
     # Apply the inverse operators and check the result
-    for inv in [inv_1, inv_2]:
+    for inv in [inv_1, inv_2, inv_4]:
         for method in ['MNE', 'dSPM', 'sLORETA']:
```
Comment (agramfort) [inline mne/minimum_norm/tests/test_inverse.py]: you probably don't need to test MNE, dSPM and sLORETA here.

I want to limit the test time

**Comment by agramfort** [comment]
thx @kingjr

**Pull request merged**